### ISSUE TITLE
Changes needed: ARIA6: Using aria-label to provide labels for objects

### ISSUE THREAD

**Issue by spanchang** [w3c/wcag#1987]
Two changes suggested: one to Description and one to Test Procedure.
Description: Suggested addition:
This attribute should be used  in one of two situations:
- the text that appears to label the object or convey its purpose visually cannot be associated using native  methods offered by the technology to assign an accessible name for the object 
- the purpose of the object is evident by its context, presentation, non-visual cues and it has no text identifier 

Suggested Tests Procedure For each object where a aria-label attribute is present.
Situation A. Where text that functions as the object's visible label is present
1. The underlying technology does not offer method to natively assign aaccessible name for the object
2. The value of aria-label begins with or only includes the text that functions as the visible label for the object
Expected Results
Both 1 and 2 are true
Situation B:
The purpose of the object is indicated solely by its context and visual appearance without a text identifier   
1. The value of aria-label attribute properly describes the purpose of the object in the context in which it is presented
Expected Results
Item 1 is true

Thanks,
Sailesh

**Comment by patrickhlauke** [comment]
> 1. The underlying technology does not offer method to natively assign aaccessible name for the object

I would say (as discussed on mailing list previously) that this point is irrelevant. Are you suggesting that auditors should *fail* content if this is not the case? If so I'd strongly disagree on having this point included.

**Comment by spanchang** [comment]
No, I am not suggesting  that "auditors should fail content if this is not the case". 
What it means is that the  technique is not warranted  in the situation.
That is not what test procedure for a technique is meant for. Not passing the test procedure means the technique is not implemented correctly. 
Please see existing description for ARIA14.
I am trying to introduce consistency across the three aria-label techniques: ARIA6 / ARIA8 / ARIA14. 
In the email I suggested: consider combining the three  techniques. 
Thanks,
Sailesh

**Comment by patrickhlauke** [comment]
You're trying to only say "this technique is valid, but only in cases where the underlying technology doesn't already have a native way to assign accessibility" one way or another, it seems. The thinking seems to be the same. Without the addition of "1.", authors can choose (depending on their specific situation) to take inspiration from this technique in their work (if, for whatever reason, they're unable to use the native accessible name assignment...maybe their CMS is locked down? maybe they're not in control of various aspects of the page, just a tiny fraction? etc). By somehow suggesting that this technique only works in cases where there's no native way of doing it, you're trying to social-engineer them away from finding a solution that, in the end when done correctly, still leads to the correct end result for end users.
